tokenizer.js

```diff
@@ -1,37 +1,19 @@
-class Tokenizer {
-    constructor(str){
-        this.queue = [];
-        this.seperator = ["[", "]", "{", "}", ",", ":"];
-        this.tokenize(str);
-    }
-
-    tokenize(str, temp = []){
-        if (!str.length) return
-        
-        if (this.isQuoteClosed(temp)) temp = temp.filter(e => e !== ' ');
-
-        if (this.seperator.includes(str[0]) && this.isQuoteClosed(temp)) {
-            if (temp.length > 0) this.queue.push(temp.join('').trim());
-            this.queue.push(str[0]);
-            return this.tokenize(str.substring(1), []);
-        }
-
-        else return this.tokenize(str.substring(1), [...temp, str[0]]);
-    }
-
-    isQuoteClosed(tokenArr){
-        return tokenArr.filter(e => e === `"`).length !== 1;
-    }
-
-    getToken(){
-        return this.queue;
+const tokenize = (str, temp = [], queue = []) => {
+    const seperator = ["[", "]", "{", "}", ",", ":"];
+    if (!str.length) return queue;
+    
+    if(str[0] === ' ' && isQuoteClosed(temp))
+        return tokenize(str.substring(1), temp, queue);
+
+    if (seperator.includes(str[0]) && isQuoteClosed(temp)) {
+        if (temp.length > 0) queue.push(temp.join('').trim());
+        queue.push(str[0]);
+        return tokenize(str.substring(1), [], queue);
+    } else {
+        return tokenize(str.substring(1), [...temp, str[0]], queue);
     }
```

Comment:
P5
else절이 없어도 되겠네요.

```suggestion
      return tokenize(str.substring(1), [...temp, str[0]], queue);
```